azure_batch/main.tf

```diff
@@ -0,0 +1,183 @@
+  name                     = "batch${terraform.workspace}storage"
+  resource_group_name      = var.resource_group_name
+  location                 = var.resource_group_location
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+}
+
+# Create a container to store the files related to azure Batch
+resource "azurerm_storage_container" "input_files" {
+  name                  = var.container_name
+  storage_account_name  = azurerm_storage_account.batch_computing_s_a.name
+  container_access_type = "private"
+}
+
+# Retention policy of 1 day in the previously created container
+resource "azurerm_storage_management_policy" "files_retention_duration" {
+  storage_account_id = azurerm_storage_account.batch_computing_s_a.id
+
+  rule {
+    name    = "delete-older-than-n-days"
+    enabled = true
+    filters {
+      prefix_match = ["${var.container_name}/"]
+      blob_types   = ["blockBlob"]
+    }
+    actions {
+      base_blob {
+        delete_after_days_since_modification_greater_than = var.keep_inp_files_during_n_days
+      }
+      snapshot {
+        delete_after_days_since_creation_greater_than = var.keep_inp_files_during_n_days
+      }
+    }
+  }
+}
+
+# Create a container registry
+resource "azurerm_container_registry" "acr" {
+  name                = "batchComputingContainerRegistry"
+  resource_group_name = var.resource_group_name
+  location            = var.resource_group_location
+  sku                 = "Basic"
+  admin_enabled       = true
+}
+
+# Push a Docker image to the registry
+resource "null_resource" "push_to_acr" {
+  provisioner "local-exec" {
+    command = <<EOF
+          az acr login --name "${azurerm_container_registry.acr.login_server}" --username "${azurerm_container_registry.acr.admin_username}" --password "${azurerm_container_registry.acr.admin_password}"
+          docker tag ${var.image_name}:${var.image_tag} "${azurerm_container_registry.acr.login_server}"/${var.image_name}:${var.image_tag}
+          docker push "${azurerm_container_registry.acr.login_server}"/${var.image_name}:${var.image_tag}
+  EOF
+  }
+}
+
+
+# Create an Azure Batch account
+resource "azurerm_batch_account" "batch_account" {
+  name                 = "batchaccount${terraform.workspace}"
+  resource_group_name  = var.resource_group_name
+  location             = var.resource_group_location
+  pool_allocation_mode = "BatchService"
+  storage_account_id   = azurerm_storage_account.batch_computing_s_a.id
+}
+
+# Create the pool within the Azure Batch account
+resource "azurerm_batch_pool" "dynamic_pool" {
+  name                = "pool-${terraform.workspace}"
+  resource_group_name = var.resource_group_name
+  account_name        = azurerm_batch_account.batch_account.name
+  display_name        = "Time-based autoscaled pool"
+  vm_size             = var.vm_size
+  node_agent_sku_id   = "batch.node.ubuntu 20.04"
+
+  auto_scale {
+    evaluation_interval = "PT5M"
```

Comment:
Is this the minimum evaluation interval?

Reply:
Unfortunately yes, the lower bound on the evaluation interval is 5 minutes.